server/src/operators/dataset_operator.rs

```diff
@@ -629,3 +630,41 @@ pub async fn get_dataset_usage_query(
 
     Ok(dataset_usage)
 }
+
+pub async fn get_tags_in_dataset_query(
+    dataset_id: uuid::Uuid,
+    page: i64,
+    page_size: i64,
+    pool: web::Data<Pool>,
+) -> Result<(Vec<TagsWithCount>, i64), ServiceError> {
+    use crate::data::schema::chunk_metadata_tags::dsl as chunk_metadata_tags_columns;
+    use crate::data::schema::dataset_tags::dsl as dataset_tags_columns;
+
+    let mut conn = pool.get().await.unwrap();
+
+    let items = dataset_tags_columns::dataset_tags
+        .inner_join(chunk_metadata_tags_columns::chunk_metadata_tags)
+        .group_by(dataset_tags_columns::tag)
+        .select((
+            dataset_tags_columns::tag,
+            count(chunk_metadata_tags_columns::chunk_metadata_id),
+        ))
+        .order_by(count(chunk_metadata_tags_columns::chunk_metadata_id).desc())
```

Comment:
I'm worried about perf, but we'll test on HN. 

I'm pretty sure this forces it to calc the count for every tag in the whole dataset.